pandas/tests/scalar/timestamp/test_unary_ops.py

```diff
@@ -132,6 +133,43 @@ def test_floor(self):
         expected = Timestamp('20130101')
         assert result == expected
 
+    @pytest.mark.parametrize('timestamp', [
+        '2018-01-01 0:0:0.124999360',
+        '2018-01-01 0:0:0.125000367',
+        '2018-01-01 0:0:0.125500',
+        '2018-01-01 0:0:0.126500',
+        '2018-01-01 12:00:00',
+        '2019-01-01 12:00:00',
+    ])
+    @pytest.mark.parametrize('freq', [
+        '2ns', '3ns', '4ns', '5ns', '6ns', '7ns',
+        '250ns', '500ns', '750ns',
+        '1us', '19us', '250us', '500us', '750us',
+        '1s', '2s', '3s',
+        '1D',
+    ])
+    def test_round_int64(self, timestamp, freq):
+        """check that all rounding modes are accurate to int64 precision
+           see GH#22591
+        """
+        dt = Timestamp(timestamp)
+        unit = to_offset(freq).nanos
+        # test floor
+        result = dt.floor(freq)
+        assert result.value % unit == 0, "floor not a %s multiple" % (freq, )
+        assert 0 <= dt.value - result.value < unit, "floor error"
+        # test ceil
+        result = dt.ceil(freq)
```

Comment:
are there round errors in timedelta ops as well?

Reply:
Yes, also `timedeltas` are affected by rounding errors, but it is very unlikely that these will be found in real world applications.

BTW I found #22591 processing real scientific data, in which I had timestamps for 2018 measurements with about 1 µs resolution. On the contrary I cannot imagine of an application in which you have 50 years time deltas with such a small resolution.

Of course you could create a round-trip example:
```
Timedate -> Timedelta -> round() -> Timedate 
```
which gives different results with respect to 
```
Timedate -> round().
```
Nevertheless I would suggest to postpone the resolution of the timedelta issue after this PR is merged.

(Please note also that the current complex machinery, which tries to improve rounding errors, is implemented only for Timedate and not for Timedelta.)

Reply:
no i get it, that's why I asked. Ok to fix later, esp if its not very common.